## output: html_document

# LimmaVoom Astrocytes - position+condition

## Libraries

```{r setup, message=FALSE, warning=FALSE}
library(limma)
library(edgeR)
```

# 1. Load and prepare the data

```{r Loading data}

base_dir <- "F:/cosmx_data/broad_single_cell/differential_expression/position+condition/Astrocytes"
# load em 
counts <- read.table(file.path(base_dir, "em_Astrocytes.tsv"), header = T, sep = "\t", row.names = 1)
# load ss
ss <- read.table(file.path(base_dir, "ss_Astrocytes.tsv"), header = T, sep = "\t", row.names = 1)
```

Prepare the ss

```{r}
ss$mouse_id <- factor(ss$mouse_id)
ss$condition <- factor(ss$condition, levels = c("C", "A"))
ss$position <- factor(ss$position, levels = c("ant", "pos"))

```

# 2. Create DGEList object for normalisation

```{r}

# Create DGEList object for normalization
dge <- DGEList(counts = counts)
dge <- calcNormFactors(dge)  # Normalize library sizes

```

Filter low expressed genes

```{r}
# Define threshold
min_cpm <- 1  # Minimum CPM value for a gene to be considered expressed
min_samples <- floor(ncol(dge$counts) * 0.25)  # Gene must be expressed in at least 25% of samples

# Identify genes to keep
keep_genes <- rowSums(cpm(dge) >= min_cpm) >= min_samples

# Apply filtering
dge <- dge[keep_genes, , keep.lib.sizes = FALSE]  # Also reset library sizes

# Check how many genes remain
dim(dge)  # Number of genes left

```

# 3. Create the design matrix for fixed effects

This models both main effects (Condition, Position) and their interaction (Condition:Position).

```{r}
design <- model.matrix(~ position * condition , data = ss)

```

# 4. Apply voom transformation

voom() stabilizes variance for RNA-seq count data before running linear models.

```{r}
v <- voom(dge, design, plot = TRUE)

```

# 5. Model mouse_id as a random effect

Since Mouse_ID introduces mouse-specific variation, we compute duplicate correlation to estimate intra-mouse correlation:

```{r}
dupcor <- duplicateCorrelation(v, design, block = ss$mouse_id)

```

This step estimates how much variance is explained by Mouse_ID and adjusts for within-mouse correlations.

# 6. Fit the linear Model with mouse_id as a Random Effect

```{r}
fit <- lmFit(v, design, block = ss$mouse_id, correlation = dupcor$consensus)

```

Now, Mouse_ID is properly accounted for as a random effect.

# 7. Compute the Differential Expression

Apply Empirical Bayes smoothing for statistical testing:

```{r}
fit <- eBayes(fit)

```

# 8. Extract DE results

coeficients

```{r}
colnames(design)
```

(a) Condition effect (C vs A)

```{r}
res_condition <- topTable(fit, coef = "conditionA", number = Inf, adjust = "fdr")

```

(b) Position effect (ant vs pos)

```{r}
res_position <- topTable(fit, coef = "positionpos", number = Inf, adjust = "fdr")

```

(c) Interaction effect (Condition:Position)

```{r}
res_interaction <- topTable(fit, coef = "positionpos:conditionA", number = Inf, adjust = "fdr")

```
